File: tutorial_pkg/scripts/listener.py
#!/usr/bin/env python3
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String

# Função de callback chamada sempre que uma nova mensagem é recebida no tópico
def callback(data):
    rospy.loginfo("I heard: %s", data.data)

def listener():
    # Inicializa o nó ROS com o nome 'listener_py'
    rospy.init_node('listener_py')

    # Criação de um subscritor do tópico 'chatter'
    # rospy.Subscriber(nome_do_tópico, tipo_da_mensagem, função_de_callback)
    rospy.Subscriber("chatter", String, callback)

    # Mantém o nó em execução e processa callbacks recebidas
    rospy.spin()

if __name__ == '__main__':
    listener()

File: tutorial_pkg/src/listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

// Função de callback chamada sempre que chega uma nova mensagem no tópico
void chatterCallback(const std_msgs::String& msg)
{
    ROS_INFO("I heard: [%s]", msg.data.c_str());
}

int main(int argc, char* argv[])
{
    // Inicializa o ROS
    // O argumento "listener" corresponde ao nome atribuido ao nó
    ros::init(argc, argv, "listener");

    // Criação de um NodeHandle
    // Responsável por estabelecer a comunicação com o sistema ROS
    ros::NodeHandle nh;

    // Criação de um subscritor do tópico 'chatter'
    // nh.subscribe("nome_do_tópico", tamanho_do_buffer, função_de_callback);
    ros::Subscriber subscriber = nh.subscribe("chatter", 1, chatterCallback);

    // Entra em loop para processar callbacks sempre que novas mensagens forem recebidas
    ros::spin();

    return 0;
}
